src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,6 +27,18 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
+
+        String splitAddress[] = new String[4];
+        splitAddress = address.split(",");
+        try {
+            block = new Block(splitAddress[0]);
+            street = new Street(splitAddress[1]);
+            unit = new Unit(splitAddress[2]);
+            postalCode = new PostalCode(splitAddress[3]);
+        }catch (ArrayIndexOutOfBoundsException e){
```

Comment:
This is alright, but consider the scenario where I do something like `add John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, Street 5, #02-25 t/friends t/owesMoney`. What will happen in this case? :/